File: TsinghuaNet.Eto/TsinghuaNet.Eto/Views/ConnectionDialog.xeto.vb
Imports Eto.Forms
Imports Eto.Serialization.Xaml
Imports TsinghuaNet.Helpers
Imports TsinghuaNet.Models

Public Class ConnectionDialog
    Inherits Dialog

    Private Model As ConnectionViewModel

    Public Sub New()
        XamlReader.Load(Me)
        Model = New ConnectionViewModel()
        DataContext = Model
        FindChild(Of GridView)("ConnectionView").DataStore = Model.NetUsers
        Model.RefreshNetUsers()
    End Sub

    Private Async Sub DropSelection(sender As Object, e As EventArgs)
        Dim view = FindChild(Of GridView)("ConnectionView")
        Await Model.DropAsync(view.SelectedItems.Select(Function(user As NetUser) user.Address))
    End Sub
End Class

File: TsinghuaNet.Eto/TsinghuaNet.Eto/Views/ConnectionDialog.xeto.cs
using System;
using System.Linq;
using Eto.Forms;
using Eto.Serialization.Xaml;
using TsinghuaNet.Models;
using TsinghuaNet.ViewModels;

namespace TsinghuaNet.Eto.Views
{
    public class ConnectionDialog : Dialog
    {
        private ConnectionViewModel Model;

#pragma warning disable 0649
        private GridView ConnectionView;
#pragma warning restore 0649

        public ConnectionDialog()
        {
            XamlReader.Load(this);
            DataContext = Model = new ConnectionViewModel();
        }

        private async void DropSelection(object sender, EventArgs e)
        {
            await Model.DropAsync(ConnectionView.SelectedItems.Select(user => ((NetUser)user).Address));
        }
    }
}
